build.gradle

```
@@ -76,4 +76,132 @@ project(':configurator') {
+    packageFileExtension = ['zip', 'tar.gz']
+}
+
+import java.security.MessageDigest
+
+task downloadJre(description: 'Download OpenJRE of Linux, Windows and Mac, and check their hashes') {
+
+    doFirst {
+        if(!jreDownloadDir.exists()) {
+            jreDownloadDir.mkdirs()
+        }
+        osNames.each {String sys ->
+            def index = osNames.indexOf(sys)
+            def jreFileExtension = (sys == 'windows' ? packageFileExtension[0] : packageFileExtension[1])
+            def jreFileName = jreFileNameTemplate(sys,jreFileExtension)
+            def jreFile = file("$jreDownloadDir.absolutePath/$jreFileName")
+            if(!jreFile.exists()){
+                new URL("https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u212-b03/$jreFileName")
+                        .withInputStream {
+                    jreFile.newOutputStream() << it
+                }
+                final hash = MessageDigest.getInstance('SHA-256').with {
+                    jreFile.eachByte(1024) {buff,len ->
+                        update(buff, 0, len)
+                    }
+                    digest().encodeHex() as String
+                }
+                if(hash != jreHashes[index]) {
+                    throw new RuntimeException("Hash mismatch for downloaded $sys JRE. File may have been modified.")
+                }
+            }
+        }
+        
+    }
+}
+
+task unzipJre(description: 'Unzip JREs only when the extract directory is not existing', dependsOn:downloadJre) {
+    doFirst {
+        fileTree(jreDownloadDir).matching {
+            exclude 'jre'
+        }.each { jreFile ->
+                 if(jreFile.name.endsWith('zip')) {
+                     copy {
+                         from zipTree(jreFile)
+                         into jreExtractSubDir('windows')
+                     }
+                 }
+                 else {
+                     copy {
+                         from tarTree(resources.gzip(jreFile))
+                         into jreExtractSubDir(jreFile.name.contains('mac') ? 'mac' : 'linux')
+                     }
+                 }
+        }
+    }
+}
+
+unzipJre.onlyIf {
+    !file(jreExtractDir).exists()
+}
+
+task createPackages(description: 'Create three packages for Linux, Windows and Mac', dependsOn:unzipJre) {
+    //This task also depends on other three tasks which are created by iterating osNames
+}
+
+osNames.each {String sys ->
+    def packageDir = new File("$buildDir/distributions/packages")
+    def (taskType, fileExtension, fileCompression, scriptFile) = setPackageProperties(sys)
+    def taskCLass = Class.forName(taskType)
+    def jarDir = file("$buildDir/libs")
+    def scriptDir = file("$projectDir/launcher-scripts")
+    task "packageFor$sys"(type: taskCLass){
+        mustRunAfter unzipJre
+        archiveBaseName = "admin-console-package-for-$sys"
+        extension = fileExtension
+        if(fileCompression) compression = Compression.GZIP
```

If on simple `if` statements it's still considered best practice to wrap in curly braces to avoid later coding errors:

```suggestion
        if(fileCompression) {
            compression = Compression.GZIP
        }
```